_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp

```diff
@@ -1509,12 +1509,29 @@ mfxU8 LookAheadBrc2::GetQp(const BRCFrameParams& par)
 
     return mfxU8(m_curQp);
 }
+mfxU8 GetNewQP(mfxU32 size, mfxU32 targeSize, mfxU8 curQP)
+{
+    mfxF64 qstep = 0, qstep_new = 0;
+    qstep = QSTEP[IPP_MIN(51, curQP)];
+    qstep_new = qstep * pow((mfxF64)size / targeSize, 0.8);
+    mfxU8 qp_new = QStep2QpCeil(qstep_new);
+    if (qp_new < 51 && qstep_new >(QSTEP[qp_new] + QSTEP[qp_new + 1]) / 2)
+        qp_new++;
+    return  qp_new;
+}
 mfxU8 LookAheadBrc2::GetQpForRecode(const BRCFrameParams& par, mfxU8 curQP)
 {
-    mfxU8 qp = curQP + (mfxU8)par.NumRecode;
+    mfxU8 qp = curQP;
+    if (m_maxFrameSize < par.CodedFrameSize)
+    {
+        qp = GetNewQP(par.CodedFrameSize, m_maxFrameSize, curQP);
```

Comment:
is it really possible that ```curQP``` > 51 here? I believe (according to the code on line 1529) that is not

Reply:
Min, Max QP can be user defined.